src/double.js

```diff
@@ -0,0 +1,57 @@
+const ms = require('ms');
+
+export default class DoubleDecker {
+
+  constructor (one, two) {
+    this.one = one;
+    this.two = two;
+  }
+
+  get (k, cb) {
+    return this.one.get(k, (err, res) => {
+      if (err) return cb(err);
+      if (!res) return cb(null, null);
+      if (res < Date.now()) {
+
+        // SIDEEFFECT! kickoff del -- note this is an
+        // asyhronous function, but be purposely ignore
+        // it's return and just hope it goes through...
+        this.del(this.one, this.two, k);
+        return cb(null, null);
+      }
+      return this.two.get(k, cb);
+    });
+  }
+
+  set (k, v, ttl, cb) {
```

Comment:
In general I definitely agree, I somehow make an exception for k/v sometimes, I think maybe just because I've spent too much time reading the lodash sourcecode and k/v always mean key/value in my mind. I'll leave this here, but definitely hear you on the general rule.